hibernate-core/src/main/java/org/hibernate/sql/results/graph/embeddable/AbstractEmbeddableInitializer.java

```diff
@@ -93,26 +96,27 @@ public void resolveKey(RowProcessingState rowProcessingState) {
 		// todo (6.0) : register "parent resolution listener" if the composite is defined for `@Parent`
 		//		something like:
 
-		final SingularAttributeMapping parentInjectionTarget = embeddedModelPartDescriptor.getParentInjectionAttributeMapping();
-
-		if ( parentInjectionTarget != null ) {
-			getFetchParentAccess().findFirstEntityDescriptorAccess().registerResolutionListener(
-					// todo (6.0) : this is the legacy behavior
-					// 		- the first entity is injected as the parent, even if the composite
-					//		is defined on another composite
-					owner -> {
-						if ( compositeInstance == null ) {
-							return;
+		final PropertyAccess parentInjectionPropertyAccess = embeddedModelPartDescriptor.getParentInjectionAttributePropertyAccess();
+
+		if ( parentInjectionPropertyAccess != null ) {
+			if ( getFetchParentAccess() != null ) {
```

Comment:
We might combine the above two if blocks into one. Sonar will say the cognition complexity would be improved greatly.
```
if ( parentInjectionPropertyAccess != null && getFetchParentAccess() != null ) {
    ... ...
}
```

Reply:
Thanks @NathanQingyangXu fixed it